add ForYou page, post carousel, button foryou hover functionality navigation

Polina-Sapego opened the pull request.

Polina-Sapego committed:
add ForYou page, post carousel, button foryou hover functionality navigation

Polina-Sapego committed:
add images

OlgaNaranovich commented on assets/styles/components/index.sass:
```diff
@@ -1,2 +1,3 @@
 @import "./settings"
 @import "./Navigation"
+@import "ForYou"
```
на 3х строках - 3 разных варианта формата импортируемых файлов. Нужно определиться, какой ты используешь и везде писать одинаково (для файлов, не содержащих компоненты, обычно используют формат camelCase)

OlgaNaranovich commented on assets/styles/components/settings/UserAgreement.sass:
```diff
@@ -1,8 +1,10 @@
 @import "../../common"
```
эти импорты уже не нужны, нужно убрать во всех местах, где они зависли. 
![Screenshot from 2024-11-16 10-16-57](https://github.com/user-attachments/assets/012bfbf6-05fa-4f4d-aaaf-9d9493d3ca8c)

OlgaNaranovich commented on src/components/ForYou/Carousel.tsx:
```diff
@@ -0,0 +1,96 @@
+    });
+  };
+
+  useEffect(() => {
+    const pageElements = posts.map((postItem, index) => (
+      <div
+        key={postItem.id}
+        className={`carousel-page ${index === visibleIndex ? 'active' : ''}`}
+        style={{
+          paddingLeft: index === 0 ? '210px' : 'none',
+        }}
+      >
+        <div
+          className="page"
+          style={{
+            width: index === visibleIndex ? '95%' : '90%',
+            marginTop: index === visibleIndex ? '2%' : '5%',
+            boxShadow: index === visibleIndex ? '-20px 20px 20px 0px rgba(255, 217, 25, 1)' : 'none',
+          }}
+        >
+          <div className="image-container">
+            <img src={PostCarousel} className="page-image" alt="carousel" />
+            <div className="page-content">
+              <h1>{postItem.title}</h1>
+              <h2>
+                <span style={{ marginRight: '10px' }}>
+                  {`${postItem.id}+ |`}
+                </span>
+                <span>{postItem.albumId}</span>
+              </h2>
+            </div>
+          </div>
+        </div>
+      </div>
+    ));
+
+    setPages(pageElements);
+  }, [posts, visibleIndex]);
```
!!! useEffect тут использован не по назначению. Он не занимается сборкой компонентов

OlgaNaranovich commented on src/components/ForYou/Foryou.tsx:
```diff
@@ -0,0 +1,37 @@
+function ForYou() {
+  const [posts, setPosts] = useState<IForYouItem[]>([]);
+  const location = useLocation();
+  const isMainSettingsPage = location.pathname === '/';
```
Зачем тут эта проверка?

OlgaNaranovich commented on src/components/ForYou/Foryou.tsx:
```diff
@@ -0,0 +1,37 @@
+  const location = useLocation();
+  const isMainSettingsPage = location.pathname === '/';
+
+  useEffect(() => {
+    const originalBackgroundColor = document.body.style.background;
+    document.body.style.background = '#0c0707';
+
+    return () => {
+      document.body.style.background = originalBackgroundColor;
+    };
+  }, []);
```
снова useEffect. Какую цель он тут преследует?

OlgaNaranovich commented on src/components/ForYou/Foryou.tsx:
```diff
@@ -0,0 +1,37 @@
+    };
+  }, []);
+
+  useEffect(() => {
+    fetch('https://jsonplaceholder.typicode.com/photos?_limit=10')
+      .then((res) => res.json())
+      .then((fetchedPosts) => { setPosts(fetchedPosts); });
+  }, []);
```
А вот это абсолютно правильное использование хука

OlgaNaranovich commented on src/components/Navigation.tsx:
```diff
@@ -1,10 +1,37 @@
   return (
     <nav className="navigation">
-      <img className="navigation-setting" src={Settings} alt="settings button" />
+      <Link to="settings">
```
как на счет NavLink вместо Link? Тогда можно будет использовать свойство active

OlgaNaranovich commented on src/components/Navigation.tsx:
```diff
@@ -1,10 +1,37 @@
+          className="navigation-setting"
+          src={isSettingsPage || isHoveredSettings ? Settings : SettingsDisabled}
+          alt="settings button"
+          onMouseEnter={() => setIsHoveredSettings(true)}
+          onMouseLeave={() => setIsHoveredSettings(false)}
```
это можно хендлить с помощью псевдоклассов

OlgaNaranovich commented on src/components/Settings/Settings.tsx:
```diff
@@ -26,19 +26,19 @@ function Settings() {
+            </div>
+          </>
+        )}
+        <Outlet />
```
сейчас оно тут нужно?

Polina-Sapego committed:
resolve comments

OlgaNaranovich commented on assets/styles/components/forYou/Carousel.sass:
```diff
@@ -0,0 +1,70 @@
+              padding: 5%
+              background: linear-gradient(to top, rgba($black, 0.9), transparent)
+
+  .arrow-left, .arrow-right
```
каждый селектор - с новой строчки

OlgaNaranovich commented on assets/styles/components/forYou/Foryou.sass:
```diff
@@ -0,0 +1,15 @@
+$black: #000000
```
этот файл - дубликат, и он не используется

OlgaNaranovich commented on src/components/Navigation.tsx:
```diff
@@ -1,37 +1,29 @@
         <img
-          className="navigation-setting"
-          src={isSettingsPage || isHoveredSettings ? Settings : SettingsDisabled}
+          src={isSettingsPage ? Settings : SettingsDisabled}
```
тут нужно использовать свойство NavLink isActive, тогда isSettingsPage и isMainPage не понадобятся

OlgaNaranovich commented on src/components/Navigation.tsx:
```diff
@@ -1,37 +1,29 @@
   return (
     <nav className="navigation">
-      <Link to="settings">
+      <NavLink to="settings" className="navigation-setting">
```
попробуй реализовать список меню через мапинг

OlgaNaranovich commented on src/components/forYou/Carousel.tsx:
```diff
@@ -0,0 +1,85 @@
+              }}
+            >
+              <div
+                className="page"
```
не информативный класс

OlgaNaranovich commented on src/components/forYou/Carousel.tsx:
```diff
@@ -0,0 +1,85 @@
+
+  return (
+    <div className="main-carousel">
+      <div className="window">
```
не информативный класс **window**, к тому же стараемся не использовать зарезервированные слова, типа window, object и т.д. даже для названия классов

OlgaNaranovich commented on src/components/forYou/Carousel.tsx:
```diff
@@ -0,0 +1,85 @@
+            >
+              <div
+                className="page"
+                style={{
+                  width: index === visibleIndex ? '95%' : '90%',
+                  marginTop: index === visibleIndex ? '2%' : '5%',
+                  boxShadow: index === visibleIndex ? '-20px 20px 20px 0px rgba(255, 217, 25, 1)' : 'none',
+                }}
```
лучше по условию `index === visibleIndex`  добавить класс с дополнительными свойствами

OlgaNaranovich commented on src/components/settings/Agreementtext.tsx:
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
```
давай поправим название файла на Agreement**T**ext

OlgaNaranovich commented on src/components/settings/Agreementtext.tsx:
```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+
+function Navigation() {
```
функцию нужно переименовать в соответствии с названием файла

OlgaNaranovich commented on src/components/settings/UserAgreement.tsx:
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import Agreementtext from './Agreementtext';
```
Agreement**T**ext

Polina-Sapego committed:
ff

Polina-Sapego committed:
delete doublet and update name file ForYou/index Settings/index

Polina-Sapego committed:
fix import

Polina-Sapego committed:
fix import ForYou.sass

Polina-Sapego committed:
fix import ForYou/Carousel.tsx delete forYou/Carousel

Polina-Sapego committed:
Navigation.tsx instead of SettingsPage and isMainPage use property NavLink isActive,instead of list of links use map.
Carousel.tsx rename classes visible-part-carousel-forYou forYou-carousel-item, add class forYou-carousel-item-active for active selected carousel item
AgreementText.tsx rename file, rename function AgreementText

Polina-Sapego committed:
fix import AgreementText

Polina-Sapego committed:
fix import UserAgreement.tsx and fix Carousel.tsx class forYou-carousel-item-active
